### PR TITLE
[SCB-474] add endpoint filter (only rest endpoints) for zuul integration

### PR DESCRIPTION
**Pull request by zhengyangyong**
Signed-off-by: zhengyangyong <yangyong.zheng@huawei.com>

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

Add rest filter for spring-cloud-zuul module in order to screen highway endpoints.

only XML and Faceted project problem warnings
image

### CODE REVIEW

File: spring-boot-starter/spring-boot-starter-discovery/pom.xml [hunk 1]
```diff
@@ -73,7 +73,7 @@
 		</dependency>
 		<dependency>
 			<groupId>org.apache.servicecomb</groupId>
-			<artifactId>service-registry</artifactId>
+			<artifactId>java-chassis-core</artifactId>
```
Comment (wujimin) [inline spring-boot-starter/spring-boot-starter-discovery/pom.xml]: why need change the dependency?

File: spring-boot-starter/spring-boot-starter-discovery/src/main/java/org/apache/servicecomb/springboot/starter/discovery/CseRibbonEndpointDiscoveryFilter.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.servicecomb.springboot.starter.discovery;
+
+import org.apache.servicecomb.core.filter.EndpointDiscoveryFilter;
+import org.apache.servicecomb.serviceregistry.discovery.DiscoveryContext;
+import org.apache.servicecomb.serviceregistry.discovery.DiscoveryTreeNode;
+
+public class CseRibbonEndpointDiscoveryFilter extends EndpointDiscoveryFilter {
```
Comment (wujimin) [inline spring-boot-starter/spring-boot-starter-discovery/src/main/java/org/apache/servicecomb/springboot/starter/discovery/CseRibbonEndpointDiscoveryFilter.java]: better extend from AbstractEndpointDiscoveryFilter
and createEndpoint return com.netflix.loadbalancer.Server

File: spring-boot-starter/spring-boot-starter-discovery/src/main/java/org/apache/servicecomb/springboot/starter/discovery/ServiceCombServerList.java [hunk 3]
```diff
@@ -49,13 +49,11 @@ public ServiceCombServerList() {
...
         serviceId,
         DefinitionConst.VERSION_RULE_ALL);
-    Map<String, MicroserviceInstance> servers = serversVersionedCache.data();
-    List<Server> instances = new ArrayList<>(servers.size());
-    for (MicroserviceInstance s : servers.values()) {
-      for (String endpoint : s.getEndpoints()) {
-        URIEndpointObject uri = new URIEndpointObject(endpoint);
-        instances.add(new Server(uri.getHostOrIp(), uri.getPort()));
-      }
+    ArrayList<Endpoint> endpoints = serversVersionedCache.data();
```
Comment (wujimin) [inline spring-boot-starter/spring-boot-starter-discovery/src/main/java/org/apache/servicecomb/springboot/starter/discovery/ServiceCombServerList.java]: by previous comment, here will got List<Server>, not List<Endpoint>

File: spring-boot-starter/spring-boot-starter-discovery/src/main/java/org/apache/servicecomb/springboot/starter/discovery/CseRibbonEndpointDiscoveryFilter.java (hunk 2 above)
Comment (WillemJiang) [reply spring-boot-starter/spring-boot-starter-discovery/src/main/java/org/apache/servicecomb/springboot/starter/discovery/CseRibbonEndpointDiscoveryFilter.java]: Cse is not supposed to be use here.

Comment (zhengyangyong) [reply spring-boot-starter/spring-boot-starter-discovery/src/main/java/org/apache/servicecomb/springboot/starter/discovery/CseRibbonEndpointDiscoveryFilter.java]: this package also have a set of "CseXXXX.java" in it ,we may rename them by the way？

File: spring-boot-starter/spring-boot-starter-discovery/src/main/java/org/apache/servicecomb/springboot/starter/discovery/ServiceCombServerList.java (hunk 3 above)
Comment (zhengyangyong) [reply spring-boot-starter/spring-boot-starter-discovery/src/main/java/org/apache/servicecomb/springboot/starter/discovery/ServiceCombServerList.java]: got it

**Pull request merged**